src/backend/catalog/gp_matview_aux.c

```diff
@@ -187,7 +189,12 @@ InsertMatviewAuxEntry(Oid mvoid, const Query *viewQuery, bool skipdata)
 	values[Anum_gp_matview_aux_mvname - 1] = NameGetDatum(&mvname);
 
 	values[Anum_gp_matview_aux_has_foreign - 1] = BoolGetDatum(has_foreign);
-	
+
+	matviewRel = table_open(mvoid, NoLock);
+	viewsql = nodeToString((Node *) copyObject(get_matview_query(matviewRel)));
```

Comment:
why need `copyObject` ?

Reply:
It seems the function isn't strictly necessary here, but keeping it ensures safety and consistency in case we modify actions from the rules.